packages/augur-core/source/contracts/libraries/token/ERC1155.sol

```diff
@@ -210,6 +210,9 @@ contract ERC1155 is ERC165, IERC1155 {
         internal
     {
         require(ids.length == values.length, "ERC1155: IDs and values must have same lengths");
+        if (ids.length == 0) {
```

Comment:
This prevents a potential re-entrancy on on the acceptance check of the fake ShareToken transfer.